graveler/uncommitted_diff_iterator.go

```diff
@@ -0,0 +1,96 @@
+package graveler
+
+import (
+	"context"
+	"errors"
+
+	"github.com/treeverse/lakefs/logging"
+)
+
+type uncommittedDiffIterator struct {
+	committedManager CommittedManager
+	list             ValueIterator
+	sn               StorageNamespace
+	treeID           TreeID
+	value            *Diff
+	err              error
+}
+
+func NewUncommittedDiffIterator(manager CommittedManager, list ValueIterator, sn StorageNamespace, treeItreeID TreeID) DiffIterator {
+	return &uncommittedDiffIterator{
+		committedManager: manager,
+		list:             list,
+		sn:               sn,
+		treeID:           treeItreeID,
+	}
+}
+
+func valueExistsInCommitted(ctx context.Context, committedManager CommittedManager, sn StorageNamespace, treeID TreeID, key Key) (bool, error) {
+	_, err := committedManager.Get(ctx, sn, treeID, key)
+	if errors.Is(err, ErrNotFound) {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func getDiffType(ctx context.Context, committedManager CommittedManager, sn StorageNamespace, treeID TreeID, key Key, tombstone bool) (DiffType, error) {
```

Comment:
Here as in 'valueExistsInCommitted' looks like if you switch to use methods the number of arguments you pass will reduce and it will cleaner to call from inside the iterator